### ISSUE TITLE
Explicitely make tfjs-core a dependency of tfjs-node

### ISSUE THREAD

**Issue by etiennedupont** [tensorflow/tfjs#2761]
#### TensorFlow.js version
"@tensorflow/tfjs-node": "1.5.2"

#### Browser version
node v10.15.3
yarn (pnp) 1.16.0

#### Describe the problem or feature request
"tfjs-node" requires "tfjs-core" in there: https://github.com/tensorflow/tfjs/blob/48ad61389ca1a1fc04425a0ee433e1593b4e4004/tfjs-node/src/nodejs_kernel_backend.ts#L18

So "tfjs-core" should be listed as one of its dependencies there: https://github.com/tensorflow/tfjs/blob/48ad61389ca1a1fc04425a0ee433e1593b4e4004/tfjs-node/package.json#L53

This causes a bug with yarn pnp.

#### Code to reproduce the bug / link to feature request
**package.json**:
```js
{
  "name": "test-tfjs",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "@tensorflow/tfjs-node": "1.5.2",
    "seedrandom":"~2.4.3"
  },
  "devDependencies": {
    "@babel/cli": "7.8.4",
    "@babel/core": "7.8.4",
    "@babel/preset-env": "7.8.4"
  },
  "babel":{
    "presets": ["@babel/preset-env"],
    "plugins": []
  },
  "scripts": {
    "build": "babel src -d lib"
  },
  "installConfig": {
    "pnp": true
  }
}
```

**src/index.js**
```js
import * as tf from '@tensorflow/tfjs-node'

(() => {
    const tensor = tf.tensor([1,1])
    console.log(`tensors = ${tensor}`)
})()
```

Running:
`yarn install && yarn build && yarn node lib`

throws:

```
yarn install v1.16.0
info No lockfile found.
[1/5] 🔍  Resolving packages...
[2/5] 🚚  Fetching packages...
[3/5] 🔗  Linking dependencies...
[5/5] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 12.26s.
yarn run v1.16.0
$ babel src -d lib
Successfully compiled 1 file with Babel.
✨  Done in 0.66s.
yarn node v1.16.0
/Users/etiennedupont/Code/test-tfjs/.pnp.js:4636
    throw firstError;
    ^

Error: Package "@tensorflow/tfjs-node@1.5.2" (via "/Users/etiennedupont/Code/test-tfjs/.pnp/unplugged/npm-@tensorflow-tfjs-node-1.5.2-9c85040c75a851e8e00d9a2434e6c961b9fda177/node_modules/@tensorflow/tfjs-node/dist/nodejs_kernel_backend.js") is trying to require the package "@tensorflow/tfjs-core" (via "@tensorflow/tfjs-core") without it being listed in its dependencies (@tensorflow/tfjs, adm-zip, google-protobuf, https-proxy-agent, node-pre-gyp, progress, rimraf, tar, @tensorflow/tfjs-node)
    at makeError (/Users/etiennedupont/Code/test-tfjs/.pnp.js:55:17)
    at Object.resolveToUnqualified (/Users/etiennedupont/Code/test-tfjs/.pnp.js:4374:17)
    at Object.resolveRequest (/Users/etiennedupont/Code/test-tfjs/.pnp.js:4445:31)
    at Function.Module._resolveFilename (/Users/etiennedupont/Code/test-tfjs/.pnp.js:4627:30)
    at Function.Module._load (/Users/etiennedupont/Code/test-tfjs/.pnp.js:4543:31)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/Users/etiennedupont/Code/test-tfjs/.pnp/unplugged/npm-@tensorflow-tfjs-node-1.5.2-9c85040c75a851e8e00d9a2434e6c961b9fda177/node_modules/@tensorflow/tfjs-node/dist/nodejs_kernel_backend.js:67:11)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
error Command failed.
Exit code: 1
Command: /Users/etiennedupont/.nvm/versions/node/v10.15.3/bin/node
Arguments: lib
Directory: /Users/etiennedupont/Code/test-tfjs
Output:

info Visit https://yarnpkg.com/en/docs/cli/node for documentation about this command.
```

**Comment by rthadur** [comment]
@etiennedupont as mentioned in #2762 , we don't need to explicitly specify tfjs-core in package.json

**Comment by etiennedupont** [comment]
I know that this works outside of the scope of yarn pnp, but I don't think that relying on transitive dependencies rather than specifying the actual dependencies is the best behaviour, ask @arcanis  what he thinks about that.

By the way, the issue is common and referenced in yarn's website:
https://yarnpkg.com/advanced/migration#a-package-is-trying-to-access-another-package-

**Comment by arcanis** [comment]
Correct. Don't forget that this isn't a PnP issue - unlisted dependencies may cause existing package managers like npm or Yarn 1.x to generate invalid hoisting, breaking projects in very subtle and hard-to-debug ways. Every time you hear someone wondering why they have the wrong version of a package in their node_modules, this is likely the reason.

**Comment by nsthorat** [comment]
Thanks for the explanation! We'll merge the change.

**Comment by lina128** [comment]
Thank you @etiennedupont for the contribution and @arcanis for the explanation of unlisted dependencies. We merged the PR #2762.
